src/db/gc/snapshot.rs

```diff
@@ -162,12 +194,25 @@ where
         ));
         temp_path.persist(&target_path)?;
         tracing::info!("exported lite snapshot at {}", target_path.display());
-        *self.exported_chain_head.write() = Some(head_ts);
         *self.blessed_lite_snapshot.write() = Some(target_path);
+        *self.exported_head_key.write() = Some(head_ts.key().clone());
 
         if let Err(e) = self.reboot_tx.send(()) {
             tracing::warn!("{e}");
         }
+
+        *self.memory_db_head_key.write() = db.heaviest_tipset_key().ok();
+        tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
Why 1s? Why do we need to `sleep`?

Reply:
This was to ensure `write_ops_rx` to be fully read but it was hacky. [This commit](https://github.com/ChainSafe/forest/pull/5640/commits/72a6a39d2ca2f1504f0cf3a15a03b37ef66ac132) refactors the logic and removes the sleep hack.